src/table/use-column-widths.tsx

```diff
@@ -80,6 +82,36 @@ export function ColumnWidthsProvider({ visibleColumns, resizableColumns, childre
     }
   };
 
+  // Update getContainer when deps change to trigger new observation.
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const getContainer = useCallback(() => containerRef.current, [resizableColumns, visibleColumns, columnWidths]);
```

Comment:
`visibleColumns` is not a stable value, so all your memoization efforts are futile.

Acceptable for this PR, because some other code already uses it too, but you just made it harder to fix when it becomes a real problem

Reply:
It have to be in the dependency array nevertheless. If the getContainer was stable, the onObserve function would capture the variables just once and never update those again.

Reply:
This is related to the comments below. I think resize observer callback should not use `visibleColumns` value, because it has access to the DOM and knows actually displayed columns

Reply:
Great call out. I haven't realised that in this version I am not using column.width. I will update the code to get the last column element from the existing reference.